tests/integration/targets/lambda/tasks/main.yml

```
@@ -315,49 +315,49 @@
     check_mode: yes
     vars:
       ansible_python_interpreter: '{{ botocore_virtualenv_interpreter }}'
-  - name: lambda_info | Assert successfull retrieval of all information
+  - name: lambda_info | Assert successfull retrieval of all information 1
     assert:
       that:
       - lambda_infos_all is not failed
       - lambda_infos_all.functions | length > 0
-      - lambda_infos_all.functions[0].function_name == lambda_function_name
-      - lambda_infos_all.functions[0].runtime == lambda_python_runtime
-      - lambda_infos_all.functions[0].description == ""
-      - lambda_infos_all.functions[0].function_arn is defined
-      - lambda_infos_all.functions[0].handler == lambda_python_handler
-      - lambda_infos_all.functions[0].versions is defined
-      - lambda_infos_all.functions[0].aliases is defined
-      - lambda_infos_all.functions[0].policy is defined
-      - lambda_infos_all.functions[0].mappings is defined
-      - lambda_infos_all.functions[0].tags is defined
-      - lambda_infos_all.functions[0].architectures == ['arm64']
+      - lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | length == 1
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).runtime == lambda_python_runtime
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).description == ""
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).function_arn is defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).handler == lambda_python_handler
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).versions is defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).aliases is defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).policy is defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).mappings is defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).tags is defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).architectures == ['arm64']
```

```suggestion
      - tmp_function_info.architectures == ['arm64']
    vars:
      tmp_function_info: lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first
```

When the link starts getting this long I sometimes use a temporary variable to try and make things easier to follow